Lib/test/test_clinic.py

```diff
@@ -813,6 +813,20 @@ def test_other_bizarre_things_in_annotations_fail(self):
         )
         self.assertEqual(s, expected_failure_message)
 
+    def test_kwarg_splats_disallowed_in_function_call_annotations(self):
+        expected_error_msg = (
+            "Error on line 0\n"
+            "Cannot use a kwarg splat in a function-call annotation"
+        )
+        dataset = (
+            'module fo\nfo.barbaz\n   o: bool(**{None: "bang!"})',
+            'module fo\nfo.barbaz -> bool(**{None: "bang!"})',
+        )
```

Comment:
No change requested here, but just wanted to mention: this bug doesn't actually have anything to do with whether or not the keys in the dictionary being unpacked are `None` or not. The parsed AST would have an `ast.keyword(arg=None)` node even if the dictionary being unpacked only had string keys. The reason why `arg` is set to `None` on the `ast.keyword` node when the `ast` module parses function calls such as `f(**{"foo": "bar"})` is to distinguish the keyword-argument splat from other kinds of keyword arguments passed to function calls, e.g. `f(foo=42)`

Reply:
I missed that point. Thanks! And yeah, no need to change the test code.

Reply:
For completeness:

```pycon
>>> import ast
>>> call_node_1 = ast.parse('func(foo=42)').body[0].value
>>> call_node_1.keywords[0].arg
'foo'
>>> print(ast.dump(call_node_1, indent=2))
Call(
  func=Name(id='func', ctx=Load()),
  args=[],
  keywords=[
    keyword(
      arg='foo',
      value=Constant(value=42))])
>>> call_node_2 = ast.parse('func(**{"foo": 42})').body[0].value
>>> call_node_2.keywords[0].arg is None
True
>>> print(ast.dump(call_node_2, indent=2))
Call(
  func=Name(id='func', ctx=Load()),
  args=[],
  keywords=[
    keyword(
      value=Dict(
        keys=[
          Constant(value='foo')],
        values=[
          Constant(value=42)]))])
```